sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/EditImmunizationSteps.java

```diff
@@ -344,6 +344,49 @@ public EditImmunizationSteps(
               "Additional details text area is enabled");
           softly.assertAll();
         });
+    When(
+        "I check the specific created data is correctly displayed on Edit immunization page",
+        () -> {
+          webDriverHelpers.waitUntilIdentifiedElementIsPresent(UUID);
+          collectedImmunization = collectSpecificImmunizationData();
+          createdImmunization = CreateNewImmunizationSteps.immunization;
+          ComparisonHelper.compareEqualFieldsOfEntities(
+              collectedImmunization,
+              createdImmunization,
+              List.of("dateOfReport", "responsibleRegion", "responsibleDistrict"));
+        });
+    Then(
+        "^I check that Immunization data is displayed as read-only on Edit immunization page$",
+        () -> {
+          webDriverHelpers.waitForPageLoadingSpinnerToDisappear(30);
+          TimeUnit.SECONDS.sleep(15);
```

Comment:
why this long sleep?

Reply:
Somethimes when the spinner dissapear it takes time for the elements to be clickable. But you're right 5 seconds should be enough.

Reply:
done